frame/staking/src/pallet/mod.rs

```diff
@@ -1058,14 +1096,15 @@ pub mod pallet {
 			ensure!(!targets.is_empty(), Error::<T>::EmptyTargets);
 			ensure!(targets.len() <= T::MaxNominations::get() as usize, Error::<T>::TooManyTargets);
 
-			let old = Nominators::<T>::get(stash).map_or_else(Vec::new, |x| x.targets.into_inner());
-
+			let old = NominatorsHelper::<T>::get_any(stash).map(|n| n.targets).unwrap_or_default();
 			let targets: BoundedVec<_, _> = targets
 				.into_iter()
 				.map(|t| T::Lookup::lookup(t).map_err(DispatchError::from))
 				.map(|n| {
 					n.and_then(|n| {
```

Comment:
Can't this logic with adding targets be abstracted to `do_add_nominator`? Should be able to reduce some code redundancy